### ISSUE TITLE
ObjectDisposedException with Android.EntryRenderer

### ISSUE THREAD

**Issue by vinnivan** [xamarin/Xamarin.Forms#5238]
### Description
Starting with Xamarin Forms 3.5 calling entry.Focus() in on Appearing causes ObjectDisposedException to be called the second time the view is shown.

### Steps to Reproduce
1. Notice entry has focus on the first page at bottom of page
2. From Hamburger->choose About - notice entry has focus
3. From Hamburger->choose Browse - app crashes with ObjectDisposedException

### Expected Behavior
Entry should again have focus like step 1. This is how it has always behaved before v3.5

### Actual Behavior
App Crashes

### Basic Information

- Version with issue:
- Last known good version: 3.4.x
- IDE:VS2017 v15.9.6
- Platform Target Frameworks: 
  - iOS: not tested
  - Android: Android 8.0 API Level26
  - UWP:  not tested
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Reproduction Link

[EntryCrash.zip](https://github.com/xamarin/Xamarin.Forms/files/2861519/EntryCrash.zip)

**Comment by transis2** [comment]
Team, be aware that we are having this problem with focus not only with Entry, but also with Listview Focus.

System.ObjectDisposedException: Cannot access a disposed object. Object name: 'Xamarin.Forms.Platform.Android.ListViewRenderer'.

And Unfocus is faccing the same problem:

Java.Interop
JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self)
Java.Interop
JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters)
Android.Views
View.ClearFocus ()

Xamarin.Forms
VisualElement.Unfocus ()
TransisMobileX.View.Gerais
TelaVenda.TXT_COD_ITEM_UnFocus (

**Comment by luveti** [comment]
I too am having this issue. Here is the minimal code used to reproduce it:

```csharp
using Xamarin.Forms;

namespace AndroidEntryRendererDisposeIssue {
    public class App : Application {
        protected override void OnStart() {
            var entry = new Entry();

            var button = new Button { Text = "Focus Entry" };
            button.Clicked += (sender, e) => entry.Focus();

            var grid = new Grid {
                HorizontalOptions = LayoutOptions.FillAndExpand,
            };
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
            grid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
            grid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
            grid.Children.Add(entry, 0, 0);
            grid.Children.Add(button, 0, 1);

            MainPage = new ContentPage {
                Content = grid
            };

            grid.Children.Clear();
            grid.Children.Add(entry, 0, 0);
            grid.Children.Add(button, 0, 1);
        }
    }
}
```
Here's a zip of the project:
[AndroidEntryRendererDisposeIssue.zip](https://github.com/xamarin/Xamarin.Forms/files/2881029/AndroidEntryRendererDisposeIssue.zip)

**Comment by samhouts** [comment]
Duplicate of #5202

**Comment by andreinitescu** [comment]
Good news, it looks fixed with 4.0.0.135214-pre4, would be helpful for all which reported to confirm

**Comment by vinnivan** [comment]
I was able to verify with my test app that I'm no longer able to reproduce the error after upgrading to 4.0.0.13524-pre4.
